packages/realm/src/Realm.ts

```diff
@@ -992,6 +1050,30 @@ export class Realm {
   ): ClassHelpers {
     return this.classes.getHelpers<T>(arg);
   }
+
+  /**
+   * Update subscriptions with the initial subscriptions if needed.
+   *
+   * @param initialSubscriptions The initial subscriptions.
+   * @param realmExists Whether the realm already exists.
+   */
+  private handleInitialSubscriptions(initialSubscriptions: InitialSubscriptions, realmExists?: boolean): void {
+    const shouldUpdateSubscriptions = initialSubscriptions.rerunOnOpen || !realmExists;
+    if (shouldUpdateSubscriptions) {
+      this.subscriptions.updateSync(initialSubscriptions.update);
+    }
+  }
+}
+
+/**
+ * Determine whether an object is a top level asymmetric object.
+ *
+ * @param objectSchema The schema of the object.
+ * @returns `true` if it is asymmetric, otherwise `false`.
+ * @internal
+ */
+function isTopLevelAsymmetric(objectSchema: binding.ObjectSchema): boolean {
+  return objectSchema.tableType === binding.TableType.TopLevelAsymmetric;
```

Comment:
Could also just call this `isAsymmetric()`, but given that the table type in the binding is called `TopLevelAsymmetric` there may be different cases to handle later?

Reply:
I agree 👍

Reply:
+1 for `isAsymmetric`